### ISSUE TITLE
Profile resets on createDefaultProfile

### ISSUE THREAD

**Issue by MMitlmeier** [circus-kitchens/flutter_datawedge#74]
I am experiencing a similar issue as described in https://github.com/circus-kitchens/flutter_datawedge/issues/22.
Also running Android 11 and Datawedge 11.3.28. (Everything seemed to work fine on Android 10)

Every time createDefaultProfile is called the settings of the profile are reset.
I need a specific decoder to be enabled which is not enabled by default.

I also tested this with the example app from this repo and tried to use the updateProfile method to specifically enable the decoder each time I need it. However I couldn't make it work and I am not sure if the error is in my config structure.
Following this documentation https://techdocs.zebra.com/datawedge/13-0/guide/api/setconfig/ I added this code in the button_tab_view of the example app:

```dart
Row( 
  children: [
    Expanded(
      child: ElevatedButton(
        onPressed: () => fdw.updateProfile(
          profileName: 'Example app profile',
          pluginName: 'BARCODE',
          config: {
            'PARAM_LIST': {
              'scanner_selection': 'auto',
              'EXTRA_DATA': {
                'decoder_australian_postal': true,
                'decoder_i2of5': true,
              }
            }
          },
        ),
        child: Text('Enable decoder'),
      ),
    ),
  ],
),
```

The ActionResult returns 'SUCCESS' but the decoder stays disabled.

When createDefaultProfile is called there are ActionResults that return 'FAILURE RESULT_CODE: PROFILE_ALREADY_EXISTS' which seems to be correct.

And is there a way to select the active profile? Because at the moment it seems like I need to call createDefaultProfile every time (even if the profile already exists) when I want to set the active profile.

**Comment by t-unit** [comment]
I would recommend not checking for existing profiles via `requestProfiles` and not creating new profiles all the time. You still might want to call `updateProfile` every time to make sure the config is valid for your use case.

This library just forwards command via intents to the zebra device/software itself. If there are differences between Android versions, there unfortunately not much we can do.

**Comment by MMitlmeier** [comment]
Yes, I will try to avoid creating new profiles. However this comment https://github.com/circus-kitchens/flutter_datawedge/issues/22#issuecomment-1752969264 led me to believe that trying to create a profile that already exists will not override settings which does not seem to be the case.

I will try to use `requestProfiles` and `requestActiveProfile` so I can check if the profile already exists and is selected.
But what would be your recommended solution if the active profile isn't the one I need (and maybe it already exists and I can't use `createDefaultProfile`)?


Do you have any experiences with trying to enable decoders with `updateProfile` ?
I would be interested in seeing how a working `updateProfile`-call looks like.

It's not that easy to try different android versions for me as the device is managed by a different company.
It would help a lot to know if the code that I posted above works on a different version.

**Comment by t-unit** [comment]
@MMitlmeier this library used to create/overwrite a profile every time it was initialized. This not the case anymore.

> I will try to use requestProfiles and requestActiveProfile so I can check if the profile already exists and is selected.
But what would be your recommended solution if the active profile isn't the one I need (and maybe it already exists and I can't use createDefaultProfile)?

That is a bit up to you. You might decide to create new profiles you don't find a suitable one. Or update existing profiles. I'm not sure if there is a recommended way by Zebra.

> Do you have any experiences with trying to enable decoders with updateProfile ?
> I would be interested in seeing how a working updateProfile-call looks like.

Sorry, no

> It's not that easy to try different android versions for me as the device is managed by a different company.
> It would help a lot to know if the code that I posted above works on a different version.

I can't help either, I currently only have access to a Zebra with Android 10

**Comment by james-rich** [comment]
I am having a similar issue with enabling a barcode in the profile, the example keystroke works but I have read through the same documentation and cannot get the profile to enable or disable a barcode, @bobekos any chance you could provide an example of this please?

**Comment by bobekos** [comment]
@MMitlmeier 
This should work (tested on Zebra TC21 DataWedge 8.2.60)
```dart
fdw.updateProfile(
   profileName: 'Example app profile',
   pluginName: 'BARCODE',
   config: {
     'scanner_selection': 'auto',
     'decoder_australian_postal': 'true',
     'decoder_i2of5': 'true',
   },
 )
```
The config map is already parsed as the "PARAM_LIST" Bundle on the native side so you don't need to pass it extra.

@james-rich 
Which exactly option to you try to activate? Can you provide me a example based on the [documentation](https://techdocs.zebra.com/datawedge/13-0/guide/api/setconfig/).

**Comment by james-rich** [comment]
Thanks you @bobekos I was doing it a similar way to @MMitlmeier and looking at the documentation it makes sense, however your example solves my issue so I can now update the profile and I also believe it solves @MMitlmeier issue as you can now create a new profile and set the initial decodes.

**Comment by douniazed-ab** [comment]
how to run it without main.dart please ( sorry if its a stupid question i'm beginner )

**Comment by bobekos** [comment]
> how to run it without main.dart please ( sorry if its a stupid question i'm beginner )
> 

What do you mean without main.dart? Can you share your idea of what you are trying to achieve?

**Comment by t-unit** [comment]
closing due to inactivity, feel free to re-open
